pystiche_papers/ulyanov_et_al_2016/_data.py

```diff
@@ -208,12 +212,22 @@ def batch_sampler(
 
     if num_batches is None:
         if impl_params:
+            # https://github.com/pmeier/texture_nets/blob/aad2cc6f8a998fedc77b64bdcfe1e2884aa0fb3e/train.lua#L48
+            # Due to the use of optim.default_transformer_epoch_optim_loop, the num_iter is the result of
```

Comment:
This has nothing to do with `optim.default_transformer_epoch_optim_loop`, but rather that we split `num_iterations` into multiple `epochs` with corresponding `num_batches`.